cpp/src/arrow/type.cc

```diff
@@ -37,6 +37,32 @@
 
 namespace arrow {
 
+static constexpr const char* kNullFingerprint = "z";
+
+static constexpr const char* kBooleanFingerprint = "b";
+static constexpr const char* kUInt8Fingerprint = "I1";
+static constexpr const char* kUInt16Fingerprint = "I2";
+static constexpr const char* kUInt32Fingerprint = "I4";
+static constexpr const char* kUInt64Fingerprint = "I8";
+static constexpr const char* kInt8Fingerprint = "i1";
+static constexpr const char* kInt16Fingerprint = "i2";
+static constexpr const char* kInt32Fingerprint = "i4";
+static constexpr const char* kInt64Fingerprint = "i8";
+static constexpr const char* kHalfFloatFingerprint = "f2";
+static constexpr const char* kFloatFingerprint = "f4";
+static constexpr const char* kDoubleFingerprint = "f8";
+
+static constexpr const char* kStringFingerprint = "su4";
```

Comment:
`str`, `bin` are prefixes that are self-explanatory (su/sb are not), albeit one character longer.

Reply:
Good idea.